src/gnc/attitude_estimator.cpp

```diff
@@ -438,10 +439,35 @@ void attitude_estimator_reset(AttitudeEstimatorState &state,
     double t, lin::Vector3d const &r_ecef, lin::Vector3f const &b_body,
     lin::Vector3f const &s_body) {
   GNC_ASSERT_NORMALIZED(s_body);
-
-  // TODO : Implement this
-  // https://github.com/pathfinder-for-autonomous-navigation/psim/issues/187
+  
   lin::Vector4f q_body_eci = lin::nans<float, 4, 1>();
+  
+  lin::Vector3f b_eci{};
+  lin::Vector4f q{};
+  env::magnetic_field(t, lin::Vector3f(r_ecef), b_eci);  // b_eci = b_ecef
+  env::earth_attitude(t, q);                             // q = q_ecef_eci
+  utl::quat_conj(q);                                     // q = q_eci_ecef
+  utl::rotate_frame(q, b_eci);                           // b_eci = b_eci  
+  env::magnetic_field(t, lin::Vector3f(r_ecef), b_eci);
```

Comment:
Pretty sure there is also a `lin::cast<float>(r_ecef)` for something like this. It doesn't matter in this instance because the function asks for the vector by value and not a stream. However, if working with `lin::` functions it's better to use `lin::cast<float>(r_ecef)`.